node/src/node/profile.cpp

```diff
@@ -74,6 +74,14 @@ profile_t::profile_t(context_t& context, const std::string& name_):
     }
 }
 
+auto profile_t::publish_on() const -> std::uint32_t {
+    return this->as_object().at("publish-on", 0u).as_uint();
+}
+
+auto profile_t::unpublish_under() const -> std::uint32_t {
+    return this->as_object().at("unpublish-under", 0u).as_uint();
```

Comment:
Why zero btw? You can calculate more smart default value based on publish_on value.